lib/slapi/client/base.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+# Base Client for Slapi Class
+# Its main functions are to:
+#  1. Start the Bot
+#     - Set listeners and which helpers to utilize
+class Slapi
+  def self.run
+    @client.on :hello do
+      @logger.info("Slapi: Successfully connected, welcome '#{@client.self.name}' to the '#{@client.team.name}' team at https://#{@client.team.domain}.slack.com.")
+    end
+
+    @client.on :message do |data|
+      case data.text
+      when /#{bot_prefix(data)}ping/ then
+        @logger.debug("Slapi: #{data.user} requested ping")
+        ping(data)
+      when /#{bot_prefix(data)}help/ then
+        @logger.debug("Slapi: #{data.user} requested help")
+        get_help(data)
+      when /#{bot_prefix(data)}reload/ then
+        @logger.debug("Slapi: #{data.user} requested plugin reload")
+        reload(data)
+      when /#{bot_prefix(data)}/ then
+        @logger.debug("Slapi: #{data.user} request forwarded to check against plugins")
+        plugin(data)
+      end unless data.user == @client.self.id
+    end
+    @client.start_async
+  end
+
+  def self.ping(data)
+    chat(
+      data,
+      title: 'Bot Check',
+      text: 'PONG',
+      color: '#229954'
+    )
+  end
+
+  def self.get_help(data)
+    # Remove when doing level 2 help or responding with specific plugin help
+    unless (data.text.include? "#{bot_prefix(data)}help ") || (@help_options['level'] == 2)
+      help_text = "Please use `@#{@bot_name} help plugin_name` for specific info"
+    end
+    if help_verify(data)
+      chat(
+        data,
+        pretext: help_text,
+        fallback: 'Your help has arrived!',
```

Comment:
An issue should be logged to externalize interaction strings like this for internationalization.  Whether we ever get to that is another story, however there are many interaction strings and we should note that as a known issue.